core/src/main/java/com/linecorp/armeria/server/ServerPort.java

```diff
@@ -66,14 +92,17 @@ public ServerPort(InetSocketAddress localAddress, SessionProtocol protocol) {
             }
         }
 
-        requireNonNull(protocol, "protocol");
-        checkArgument(protocol == HTTP || protocol == HTTPS,
-                      "protocol: %s (expected: %s or %s)", protocol, HTTP, HTTPS);
-
+        requireNonNull(protocols, "protocols");
+        checkArgument(protocols.stream().allMatch(p -> p == HTTP || p == HTTPS || p == PROXY),
+                      "protocol: %s (expected: %s, %s or %s)", protocols, HTTP, HTTPS, PROXY);
         this.localAddress = localAddress;
-        this.protocol = protocol;
+        this.protocols = ImmutableSortedSet.copyOf(protocols);
```

Comment:
If you use `immutableEnumSet`, I don't think you need to use `ImmutableSortedSet`

Reply:
Used `Sets.immutableEnumSet` here.